src/spatialdata_plot/pl/utils.py

```diff
@@ -1379,46 +1398,56 @@ def _create_initial_element_table_mapping(
                     warnings.warn(
                         f"The element '{element}' is not annotated by table '{table_name}'", UserWarning, stacklevel=2
                     )
-                element_table_mapping[element].add(table_name)
+                if isinstance(mapping := element_table_mapping[element], set):
+                    mapping.add(table_name)
+    assert isinstance(element_table_mapping, dict)
```

Comment:
I personally dislike spurious asserts. Can you verify before and block this path preemptively? If not, rather throw an informative warning

Reply:
This is purely due to mypy not accepting anything else. The only way to get around this is by adopting the whole code base to use params per element.

Reply:
Mypy gets very confused because of the data structure constantly changing. I tried casting but in most cases this does not help